### DISCUSSION TITLE
Is it possible to see the execution time for a "View" Collection?

### DISCUSSION THREAD

**Discussion by Mashiane** [pocketbase/pocketbase#5663] [Q&A]
Hi

Is it possible to see the execution time for a view collection? For some reason I am experiencing delays especially on first execution / first run on my views on my app. Im using javascript sdk?

Id like to find out how I can optimize my queries on the views etc.

These are the slow at start views, from a 10K records table.

`SELECT id, checkindate, provinceid, COUNT(*) AS records FROM bma_beneficiaryattendance group by provinceid, checkindate`

`SELECT id, checkindate, COUNT(*) AS records FROM bma_beneficiaryattendance group by checkindate`

Can you please advise, baffling..

**Comment by ganigeorgiev** [comment]
Aggregations often require table scan and it is usually a task that is known to get slower with the more data you have (_this is not SQLite specific_).

You can find the execution time for a view query when you send request to it in the _UI > Logs_.

Alternatively you can also start the executable with `--dev` and you should see the queries and their exec time in the stderr output.

> how I can optimize my queries 

The easiest way is to try to add _individual_ indexes from the Collection options > "Unique constraints and indexes " for the fields you are grouping, filtering or sorting.

This will depend on the final query (aka. do you have `sort` or `filter` params?) and without context about your collections structure it is difficult to predict what indexes are needed and what route the SQLite query planner will take.

You can try to analyze your queries using the [sqlite3 cli](https://sqlite.org/cli.html) by running `sqlite3 ./pb_data/data.db`:

```
sqlite> EXPLAIN QUERY PLAN yourFinalQuery;

# or to see if the cli tool can suggest you an index using the .expert command
sqlite> .expert
sqlite> yourFinalQuery;
```
